py4DSTEM/visualize/show.py

```diff
@@ -258,31 +264,33 @@ def show(ar,figsize=(8,8),cmap='gray',scaling='none',clipvals='minmax',
         if max != None:
             max = np.log(max)
     elif scaling == 'power':
-        _mask = ar>0
+        _mask = ar.data>0
         _ar = np.zeros_like(ar.data,dtype=float)
-        _ar[_mask] = np.power(ar[_mask],power)
+        _ar[_mask] = np.power(ar.data[_mask],power)
+        _ar[~_mask] = -np.inf
         if min != None:
             min = np.power(min,power)
         if max != None:
             max = np.power(max,power)
     else:
         raise Exception
-    _ar = np.ma.array(data=_ar.data,mask=np.logical_and(ar.mask==False,_mask)==False)
+    
+    _ar = np.ma.array(data=_ar.data,mask=~_mask)
 
     # Set the clipvalues
     if clipvals == 'minmax':
-        vmin,vmax = np.ma.min(_ar),np.ma.max(_ar)
+        vmin,vmax = np.nanmin(_ar),np.nanmax(_ar)
```

Comment:
do the nanmin/nanmax/nanmedian etc functions play nicely with masked arrays? the old code here could break if there were non-masked nan entries, and this new code will take care of that - however, the old code also ensured we we correctly ignoring masked array elements, which np.min/np.max/np.median/etc don't do.  so, just confirming before pulling this in, will these functions handle both cases?

Reply:
Yes, I was concerned about the same thing so I did check that they respect the mask (why they're different functions, I'm not sure... and there is no such thing as `np.ma.nanmean`, which is confusing) 

The behavior is:
| Data | Mask | `np.mean` | `np.nanmean` | `np.ma.mean` |
| ----- | ------- | ---------| ----------- | ----------- |
| `[1.0, 2.0, nan, nan]` | `[False, False, False, False]` | NaN | 1.5 | 1.5 |
| `[1.0, 2.0, nan, nan]` | `[False, False, False,  True]` | NaN | 1.5 | NaN |
| `[1.0, 2.0, nan, nan]` | `[False, False, True, True]` | 1.5 | 1.5 | 1.5 |
| `[1.0, 2.0, nan, nan]` | `[False,  True, False,  True]` | NaN | 1.0 | NaN |

So `nan___` does indeed respect the mask, and handles any rogue NaNs in the input array